### PR TITLE
[ js_logger ] - creating JsLogger

### PR DESCRIPTION
**Pull request by mgrinko**
@makasim please review
Currently working on tests

### CODE REVIEW

File: Resources/config/services.xml [hunk 1]
```diff
@@ -52,5 +53,10 @@
 
         <service id="logger.exception" alias="fp_badaboom.logger">
         </service>
+
+        <service id="fp_badaboom.js_logger.twig_extension" class="%fp_badaboom.js_logger.twig_extension.class%">
```
Comment (makasim) [inline Resources/config/services.xml]: fp_badaboom.twig_extension.js_logger

File: Tests/Functional/Controller/LogControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        $this->assertEquals(
+            400,
+            $client->getResponse()->getStatusCode()
+        );
+    }
+}
+
+class TestSender extends AbstractChainNode
+{
+    public $message;
```
Comment (makasim) [inline Tests/Functional/Controller/LogControllerTest.php]: save context

File: Tests/Twig/Extension/JsExceptionLogExtensionTest.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<?php
+namespace Fp\BadaBoomBundle\Tests\JsLogger;
```
Comment (makasim) [inline Tests/Twig/Extension/JsExceptionLogExtensionTest.php]: namespace?

File: Twig/Extension/JsExceptionLogExtension.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+class JsExceptionLogExtension extends \Twig_Extension
+{
+    private $router;
+
+    public function __construct(UrlGeneratorInterface $router)
+    {
+        $this->router = $router;
+    }
+
+    public function getFunctions()
```
Comment (makasim) [inline Twig/Extension/JsExceptionLogExtension.php:16]: test for this method?

File: Controller/LogController.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+class LogController extends Controller
+{
+    public function createAction(Request $request)
+    {
+        $level = $request->query->get('level');
+        $message = $request->query->get('msg');
+        $context = $request->query->get('context', array());
+
+        if (false == isset($context['file'], $context['line'])) {
+            return new Response('', 400);
```
Comment (makasim) [inline Controller/LogController.php]: throw new HttpException(400);

File: FpBadaBoomBundle.php [hunk 6]
```diff
@@ -1,12 +1,12 @@
 <?php
 namespace Fp\BadaBoomBundle;
 
-use BadaBoom\Bridge\Psr\Logger;
-use Psr\Log\LoggerInterface;
 use Symfony\Component\HttpKernel\Bundle\Bundle;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\DependencyInjection\Definition;
 
+use BadaBoom\Bridge\Psr\Logger;
```
Comment (makasim) [inline FpBadaBoomBundle.php]: is it used?

File: Tests/Functional/Controller/LogControllerTest.php [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        $this->assertTrue($client->getResponse()->isSuccessful());
+
+        $this->assertTrue(
+            $client->getResponse()->headers->contains(
+                'Content-Type',
+                'image/gif'
+            )
+        );
+
+        $exception = $sender->context->getException();
```
Comment (makasim) [inline Tests/Functional/Controller/LogControllerTest.php]: you should add assert that context is instance of `Context`. If not there will be a fatal error.

Comment (mgrinko) [inline Tests/Functional/Controller/LogControllerTest.php]: Done

**Pull request merged**